projects/output-and-error-handling/main.go

```diff
@@ -0,0 +1,89 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+)
+
+const maxWaitSeconds = 5
+const weatherURL = "http://localhost:8080"
+
+var firstTry = true
+
+func main() {
+	res, err := getWeather()
+
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error: %v", err)
+		os.Exit(1)
+	}
+	fmt.Println(res)
+}
+
+func getWeather() (string, error) {
+	for {
+
+		res, err := http.Get(weatherURL)
+
+		if err != nil {
+			return "", fmt.Errorf("failed to fetch weather: %v", err)
```

Comment:
```suggestion
			return "", fmt.Errorf("failed to fetch weather: %w", err)
```

If you use `%w` instead of `%v`, error information gets preserved. e.g. [`errors.Is`](https://pkg.go.dev/errors#Is) will return true for the wrapped error.

Reply:
Is this how error should be handled. I mean should we use %w for every error if it is wrapped in another error to preserve error information, or can we choose the behaviour?

Reply:
Yes, I can't think of a time you'd be using `fmt.Errorf` to wrap an error where you _wouldn't_ want to use `%w`.

(The result of this would be to hide the underlying type of error, which... There are convoluted scenarios where you may want to do so, but are pretty weird :))